### PR TITLE
Add a Prometheus handler.

### PR DESCRIPTION
**Pull request by brendandburns**
This enables easy prometheus metrics for clients.

Really just the beginning of easy monitoring for this library.

### CODE REVIEW

File: src/KubernetesClient/KubernetesRequestDigest.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                    kind,
+                    hasWatchParameter(request) ? "watch" : request.Method.ToString());
+            }
+            catch (Exception)
+            {
+                return nonResource(urlPath);
+            }
+        }
+
+        private static KubernetesRequestDigest nonResource(string urlPath)
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/KubernetesClient/KubernetesRequestDigest.cs]: done.

File: src/KubernetesClient/KubernetesRequestDigest.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    }
+                }
+
+                return new KubernetesRequestDigest(
+                    urlPath,
+                    false,
+                    apiGroup,
+                    apiVersion,
+                    kind,
+                    hasWatchParameter(request) ? "watch" : request.Method.ToString());
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/KubernetesClient/KubernetesRequestDigest.cs]: done.

File: src/KubernetesClient/KubernetesRequestDigest.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            }
+        }
+
+        private static KubernetesRequestDigest nonResource(string urlPath)
+        {
+            KubernetesRequestDigest digest = new KubernetesRequestDigest(urlPath, true, "nonresource", "na", "na", "na");
+            return digest;
+        }
+
+        public static bool isResourceRequest(string urlPath)
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/KubernetesClient/KubernetesRequestDigest.cs]: done

File: src/KubernetesClient/KubernetesRequestDigest.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            }
+
+            try
+            {
+                string apiGroup;
+                string apiVersion;
+                string kind;
+
+                var parts = urlPath.Split('/');
+                var namespaced = urlPath.IndexOf("namespaces/", StringComparison.Ordinal) != -1;
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/KubernetesClient/KubernetesRequestDigest.cs]: done

File: src/KubernetesClient/PrometheusHandler.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var digest = KubernetesRequestDigest.Parse(request);
+            requests.WithLabels(request.Method.ToString()).Inc();
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/KubernetesClient/PrometheusHandler.cs]: done

**Comment by tg123** [comment]
/LTGM
build failure require rebase to fix

**Comment by brendandburns** [comment]
self-lgtm (b/c of the typo above :)

**Pull request merged**